### ISSUE TITLE
Extract URL content to markdown to the New File and insert your wiki-link

### ISSUE THREAD

**Issue by gilsondev** [trashhalo/obsidian-extract-url#1]
This issue was a suggestion. A new option that:

- Extract content of URL
- Return your markdown data and create a new note with title of URL page
- Change the selected URL with wiki-link of the new page

**Comment by trashhalo** [comment]
Great feedback! is the use case something like you have a document with bunch of links and you want to convert them all at once?

**Comment by gilsondev** [comment]
That's case works too. Other case is when need review a existent note and you see that content of this link need a new note.

**Comment by eleanorkonik** [comment]
I was coming here to request something similar, I think. I would love it if this plugin had an option to search a file in my vault for links and archive a markdown version of those links to a particular folder as a permanent archive.

For example, instead of having to manually create each file and add the url at the top and then run a command, it would be great if I could run a command on a particular file where I’ve linked to a bunch of external sources, and then have this plugin turn `[link](url)` into `[[link title]]` where the new file is a markdown version of the url. 

Is this within scope for this plugin? I know this issue is pretty old, but if you’re open to accepting pull requests I can maybe see if anyone else is interested in building it out? I can see a lot of uses for this and have talked to a lot of people interested in functionality like this, but need less friction than currently exists.

**Comment by trashhalo** [comment]
@eleanorkonik feeling a bit star struck seeing a comment from you.

Of course I'll build that! Do you want to control the destination folder at the plugin level or would you like to be prompted when you run the command?

**Comment by eleanorkonik** [comment]
- Destination folder at the plugin level makes the most sense to me since I intend to use it for archival purposes, and if I remember from my one abortive foray into plugin development, it's also pretty easy to add as a setting. 
- Duplicate titles, the easiest thing is probably to slap a timestamp after it and throw an error message. 

🙏thank you so much! (and I hope you enjoy the Roundup!)

**Comment by eleanorkonik** [comment]
Wait holdup, I just discovered that this plugin https://github.com/liammagee/obsidian-topic-linking actually secretly does this? Just ... not as its main thing? I missed it when I originally looked at the repo for the Roundup because it's about _topic linking_ but I think it actually just parses every pdf and link!?! 

I still think it would be useful to have your plugin do this and not all the other complicated stuff from topic linking, but maybe you can get inspiration from some of their code?

**Comment by trashhalo** [comment]
woah. I'll still work on this. I think it's an interesting feature.

**Comment by eleanorkonik** [comment]
Yeah, thinking about it more and not in the middle of the night, your feature would actually be more useful for linking to articles because it would autogen the `[[new link to the parsed text of the url]]` stuff, but, definitely look at that plugin for inspiration but it's doing some wild and relevant stuff.

**Comment by trashhalo** [comment]
@eleanorkonik Done as of 0.10.0

Demo:
https://youtu.be/6W2Kul_uuPo

**Comment by eleanorkonik** [comment]
I LOVE IT
